source/docs/software/advanced-control/state-space/state-space-flywheel-walkthrough.rst

```diff
@@ -0,0 +1,123 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A flywheel can also be modeled without access to a physical robot, using information about the motors, gearing and flywheel moment of inertia. A full derivation is presented in Section 8.2.1 of  `Controls Engineering in FRC <https://file.tavsys.net/control/controls-engineering-in-frc.pdf>`__.
+
+The ``LinearSystem`` class contains methods to easily create a model of a flywheel from the flywheel's motors, gearing and moment of inertia. The moment of inertia can be calculated using CAD software or using physics. The examples used here are detailed in the flywheel example (TODO link to allwpilib example). Note that the gearing is written as output over input -- that is, if the flywheel spins slower than the motors, this number should be greater than one.
+
+.. note:: The C++ LinearSystem class uses :ref:`docs/software/basic-programming/cpp-units:The C++ Units Library` to prevent unit mixups and assert dimensionality. 
+
+.. tabs::
+
+    .. code-tab:: java
+
+        private static final double kFlywheelMomentOfInertia = 0.00032; // kg * m^2
+        private static final double kFlywheelGearing = 1.0;
+
+        private final LinearSystem<N1, N1, N1> m_flywheelPlant = LinearSystem.createFlywheelSystem(
+                DCMotor.getNEO(2),
+                kFlywheelMomentOfInertia,
+                kFlywheelGearing,
+                12.0);
+
+    .. code-tab:: cpp
+    
+        const units::kilogram_square_meter_t kFlywheelMomentOfInertia =
+            0.00032_kg_sq_m;
+        const double kFlywheelGearing = 1.0;
+
+        frc::LinearSystem<1, 1, 1> m_flywheelPlant = frc::FlywheelSystem(
+            frc::DCMotor::NEO(2), kFlywheelMomentOfInertia, kFlywheelGearing, 12_V);
+
+.. todo:: remoteliteralincludes
+
+Kalman Filters: Observing Flywheel State
+----------------------------------------
+
+Kalman filters are used to filter our velocity measurements using our state-space model. As our flywheel model is linear, we can use a Kalman filter to estimate the flywheel's velocity. WPILib's Kalman filter takes a ``LinearSystem`` (which we found above), along with standard deviations of model and sensor measurements . We can adjust how "smooth" our state estimate is by adjusting these weights. Larger state standard deviations will cause the filter to "distrust" our state estimate and favor new measurements more highly, while larger measurement standard deviations will do the opposite. In the case of a flywheel we start with a state standard deviation of 3rad/s and a measurement standard deviation of 0.01rad/s. These values are up to the user to choose -- these weights produced a filter that was tolerant to some noise but whose state estimate quickly reacted to external disturbances for *a* flywheel -- and should be tuned to create a filter that behaves well. The following chart shows different kalman filter's response to noise and external disturbances
```

Comment:
```suggestion
Kalman filters are used to filter our velocity measurements using our state-space model. As our flywheel model is linear, we can use a Kalman filter to estimate the flywheel's velocity. WPILib's Kalman filter takes a ``LinearSystem`` (which we found above), along with standard deviations of model and sensor measurements . We can adjust how "smooth" our state estimate is by adjusting these weights. Larger state standard deviations will cause the filter to "distrust" our state estimate and favor new measurements more highly, while larger measurement standard deviations will do the opposite. In the case of a flywheel we start with a state standard deviation of 3rad/s and a measurement standard deviation of 0.01rad/s. These values are up to the user to choose -- these weights produced a filter that was tolerant to some noise but whose state estimate quickly reacted to external disturbances for *a* flywheel -- and should be tuned to create a filter that behaves well. The following chart shows different Kalman filter's response to noise and external disturbances
```

Reply:
The diff didn't work. "kalman" needs to be capitalized.

Reply:
Is this better? "Remember that the states we choose will be driven towards their :term:`reference` by the feedback controller, for which our :ref:`Linear-Quadratic Regulator <docs/software/advanced-control/state-space/state-space-intro:The Linear-Quadratic Regulator>` will pick the optinal :term:`gain`."